### PR TITLE
add parser notes

### PR DESCRIPTION
**Pull request by jievince**
Subtask of https://github.com/vesoft-inc/nebula/issues/3677

### CODE REVIEW

File: src/parser/parser.yy [hunk 1]
```diff
@@ -402,6 +405,18 @@ static constexpr size_t kCommentLengthLimit = 256;
 %type <boolval> opt_with_properties
 %type <boolval> opt_ignore_existed_index
 
+/* Define precedence and associativity of tokens.
```
Comment (Shylock-Hg) [inline src/parser/parser.yy:408]: Single line comment, `//`

Comment (jievince) [reply src/parser/parser.yy]: OK

**Review by Aiee — APPROVED** [review]

File: src/parser/scanner.lex [hunk 2]
```diff
@@ -19,6 +22,14 @@ static constexpr size_t MAX_STRING = 4096;
 
 %}
 
+/* Define some exclusive states.
```
Comment (Shylock-Hg) [inline src/parser/scanner.lex:25]: ditto.

File: src/parser/scanner.lex [hunk 3]
```diff
@@ -60,6 +71,15 @@ LABEL_FULL_WIDTH            {CN_EN_FULL_WIDTH}{CN_EN_NUM_FULL_WIDTH}*
 
 %%
 
+ /* Flex rules section */
```
Comment (Shylock-Hg) [inline src/parser/scanner.lex:74]: ditto.

File: src/parser/scanner.lex (hunk 2 above)
Comment (jievince) [reply src/parser/scanner.lex:25]: Flex doesn't allow `//` used here.

File: src/parser/scanner.lex (hunk 3 above)
Comment (jievince) [reply src/parser/scanner.lex:74]: Flex doesn't allow `//` used and, and `/*` must be indented.

**Review by Shylock-Hg — APPROVED** [review]

File: src/parser/scanner.lex [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/* Flex definitions section */
```
Comment (CPWstatic) [inline src/parser/scanner.lex:1]: Why we had different comment style in this pr?

Comment (jievince) [reply src/parser/scanner.lex:1]: These are some of the limitations of flex itself. 
The comment style `//` is forbidden by flex in the definition section of file scanner.lex.

**Review by yixinglu — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**